tool/src/js/conversions.rs

```diff
@@ -17,12 +17,19 @@ pub fn gen_value_js_to_rust(
     post_logic: &mut Vec<String>,
 ) {
     match typ {
-        ast::TypeName::StrReference => {
+        ast::TypeName::StrReference | ast::TypeName::PrimitiveSlice(_) => {
             // TODO(#61): consider extracting into runtime function
-            pre_logic.push(format!(
-                "let {}_diplomat_bytes = (new TextEncoder()).encode({});",
-                param_name, param_name
-            ));
+            if *typ == ast::TypeName::StrReference {
+                pre_logic.push(format!(
+                    "let {}_diplomat_bytes = (new TextEncoder()).encode({});",
+                    param_name, param_name
+                ));
+            } else {
+                pre_logic.push(format!(
+                    "let {}_diplomat_bytes = new Uint8Array({});",
```

Comment:
question: does this work for non-u8 types too?

Reply:
Fixed.